### ISSUE TITLE
selu activations are mysterious to me - would you help me (and hopefully others) understand how to use them?

### ISSUE THREAD

**Issue by JackBurdick** [ageron/handson-ml#138]
Hi @ageron, 

Not an issue with your code, but a theoretical question related to the code posted [here](https://github.com/ageron/handson-ml/blob/master/11_deep_learning.ipynb) around ~cell 29 that I'm hoping you can help me with.

About using [selu](https://arxiv.org/pdf/1706.02515.pdf) activation [function](https://www.tensorflow.org/versions/master/api_docs/python/tf/nn/selu). I understand that the input should be centered around 0 with a std of 1, but I’m having trouble understanding how this should be performed and how to apply this scaling to other problems.

Currently, the linked example, the mean/stds for mnist appear to be calculated pixel-wise using;
> `mnist.train.images.mean(axis=0, keepdims=True)`

which output the shape;

> `(1, 784)`

I think I can understand why that makes sense for this problem (only one channel, small input (28x28), mostly centered data) but then I have a theoretical question about performing this type of scaling on larger, more varied, inputs -- I initially assumed this should be calculated channel-wise. So for mnist I would have ended up with only one mean and one std value.

The problem I’m trying to apply this to has more than one input channel and larger dimensions, and the target may not be as well centered.  For this type of problem, would it then make sense to calculate the means/stds for each channel and apply the scaling globally to that channel?

Maybe to make this easier/more clear for others too, would you help me write an example of using a CNN with selu activation?  I'd be happy to package something together and submit a PR, but I don't understand it well enough to do it on my own yet.  Also, when using CNNs, I think I have to pay attention to [weight initialization](https://towardsdatascience.com/selu-make-fnns-great-again-snn-8d61526802a9) as well?

**TLDR**; SELU activations are a bit mysterious to me, I understand the concept, but I'm having trouble applying it in practice (correctly).  If I can figure it (input scaling, weight initialization) out, I'd be happy to help write up a quick example using CNN+SELU.

Thank you!
Jack

**Comment by ageron** [comment]
Hi Jack,
Very interesting questions, thanks. AFAIU, the self-normalizing property of the SELU activation function is only guaranteed for feedforward neural networks. Unfortunately, I doubt that it can work out of the box in CNNs, because convolutions apply locally (on small receptive fields) but the images are not normalized at that scale, only globally. I tried to find papers about building Self-Normalized CNNs, but couldn't find any.  Of course you can use the SELU activation function in a CNN, but this won't make it self-normalized.
I hope this answers your question?

**Comment by JackBurdick** [comment]
Thank you for your response -- yes, you've answered my question.

However, now I think I'm even more curious about this; I wonder if calculating means/std "on the fly" for each receptive field (prior to the activation, pre|post convolution?) would be beneficial.  If beneficial, I also wonder how much this would impact computational cost/training time.

Either way, thank you again for taking the time to answer my questions!
